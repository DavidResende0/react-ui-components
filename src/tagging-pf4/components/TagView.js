import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';
import TagCategory from './TagCategory';

import TaggingPropTypes from '../TaggingPropTypes';

class TagView extends React.Component {
  generateTagCategories = tag => (
    <li key={tag.id}>
      <TagCategory
        key={tag.id}
        tagCategory={{ id: tag.id, description: tag.description }}
        values={tag.values}
        onTagDeleteClick={this.props.onTagDeleteClick}
      />
    </li>
  );

  render() {
    const assignedTags = [...this.props.assignedTags];
    return (
      <ul className="list-inline">
        {assignedTags
          .sort(this.sortingFunction)
          .map(this.generateTagCategories)}
      </ul>
    );
  }
}
TagView.propTypes = {
  assignedTags: TaggingPropTypes.tags,
  onTagDeleteClick: PropTypes.func,
  header: PropTypes.string,
  categoryTruncate: PropTypes.func,
  valueTruncate: PropTypes.func,
  sortingFunction: PropTypes.func,

};

TagView.defaultProps = {
  header: 'Assigned tags',
  categoryTruncate: str => (str.length > 18 ? `${str.substring(0, 18)}...` : str),
  valueTruncate: str => (str.length > 18 ? `${str.substring(0, 18)}...` : str),
  sortingFunction: (a, b) => (a.description < b.description ? -1 : 1)
};

export default TagView;
