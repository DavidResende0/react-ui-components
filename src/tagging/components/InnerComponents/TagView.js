import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';
import TagCategory from './TagCategory';
import { __ } from '../../../global-functions';
import TaggingPropTypes from '../TaggingPropTypes';

class TagView extends React.Component {
  generateTagCategories = tag => (
    <li key={tag.id}>
      <TagCategory
        key={tag.id}
        tagCategory={{ id: tag.id, description: tag.description }}
        values={tag.values}
        onTagDeleteClick={this.props.onTagDeleteClick}
      />
    </li>
  );

  render() {
    const assignedTags = [...this.props.assignedTags];
    return (
      <div id="assignments_div">
        { !this.props.hideHeader &&
          <Row>
            <Col lg={12}>
              <h2>{this.props.header}</h2>
            </Col>
          </Row>}
        <Row>
          <Col lg={12}>
            <ul className="list-inline">
              {assignedTags
                .sort((a, b) => (a.description < b.description ? -1 : 1))
                .map(this.generateTagCategories)}
            </ul>
          </Col>
        </Row>
      </div>
    );
  }
}
TagView.propTypes = {
  assignedTags: TaggingPropTypes.tags,
  onTagDeleteClick: PropTypes.func,
  header: PropTypes.string,
  hideHeader: PropTypes.bool,
};

TagView.defaultProps = {
  header: __('Assigned tags'),
  hideHeader: false,
};

export default TagView;
