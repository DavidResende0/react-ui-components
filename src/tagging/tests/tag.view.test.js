import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TagView from '../components/tagView';

const assignedTags = [{ tagCategory: { description: 'Name', id: 1 }, tagValues: [{ description: 'Pepa', id: 11 }] }];
const onDelete = x => x;

describe('Tag view', () => {
  it('match snapshot', () => {
    const tree = mount(<TagView assignedTags={assignedTags} onTagDeleteClick={onDelete} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
