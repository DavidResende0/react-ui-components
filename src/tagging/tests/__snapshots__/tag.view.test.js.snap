// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag view match snapshot 1`] = `
<React.Fragment>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      lg={12}
    >
      <h2>
        Assigned tags
      </h2>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      lg={12}
    >
      <ul
        className="list-inline"
      >
        <TagCategory
          categoryTruncate={[Function]}
          key="1"
          onTagDeleteClick={[MockFunction]}
          tagCategory={
            Object {
              "description": "Name",
              "id": 1,
            }
          }
          valueTruncate={[Function]}
          values={
            Array [
              Object {
                "description": "Pepa",
                "id": 11,
              },
            ]
          }
        />
      </ul>
    </Col>
  </Row>
</React.Fragment>
`;
